### ISSUE TITLE
Game Runs Slow

### ISSUE THREAD

**Issue by nsmryan** [nsmryan/RustRoguelike#120]
Sometimes the game takes a long time between turns.

This is largely due to pathfinding on the map, which causes thousands of evaluations, even when the answers don't change much.

**Comment by nsmryan** [comment]
Considering using a single AI for collision things, if ever possible, and then only doing the rest if applicable.

**Comment by nsmryan** [comment]
I made a change that seems to have helped enormously.
Try it out and see if you detect any slow down now.

The change is in how sound is propagated- previously we used astar searches to move sound out to each square it would hit, which was hundreds or thousands of searches a turn.
Now there is a floodfill algorithm (which could be used for other things perhaps), which provides the set of positions sound can hit. The floodfill does not cross walls, so it still seems to work the same. It also does the thing where, if there is no direct path to the square, we had the sound diminish a bit.

It does need some testing- i tweaked the settings to get the same squares hit by sounds, but I also changed the distance function to use the same line drawing algorithm as other parts of the code, so other things may be slightly different and need tweaking.
